scheduling/storage_providers/memory.go

```diff
@@ -19,7 +20,40 @@ type LockInformation struct {
 	ID   string
 	Key  string
 	Date time.Time
-	Lock LockObject
+}
+
+func (l *LockInformation) Unmarshal(data []byte) error {
```

Comment:
Should move this to a lock.go file or something

Reply:
moved it to `storage_provider.go` file but can move it to `lock.go` if we want?